packages/component-library/src/components/plmg-slider/plmg-slider.tsx

```diff
@@ -0,0 +1,314 @@
+              <div class={'thumb-label-container'}>
+                <output
+                  style={this.setThumbPosition()}
+                  class={'plmg-slider-thumb-label'}
+                  htmlFor={this.name}
+                >
+                  {this.value}
+                  <span class={'plmg-thumb-triangle'} />
+                </output>
+              </div>
+            )}
+
+            <input
+              min={this.min}
+              max={this.max}
+              name={this.name}
+              value={this.value}
+              aria-valuemin={this.min}
+              aria-valuemax={this.max}
+              aria-valuenow={this.value}
+              step={this.stepValue}
+              onInput={(ev) => this.handleSliderChange(ev)}
+              style={{ background: this.setBackgroundProgressFill() }}
+              id="range"
+              type="range"
+            />
+
+            {this.marks && (
+              <div class={'labels'}>
+                {this.rangeValues.map((item, index) => (
+                  <div
+                    class={'label'}
+                    key={index}
+                    style={{ transform: this.setLabelPosition(item) }}
+                  >
+                    <span class={'tick'}>&#8205;</span>
+                    <span>{item}</span>
+                  </div>
+                ))}
+              </div>
+            )}
+          </div>
+          <label htmlfor={this.name} tabIndex={0}>
+            <div class={'plmg-slider-input-field-container'}>
+              <label htmlfor={this.name} />
+              <input
+                type={'number'}
+                name={this.name}
+                step={this.stepValue}
+                aria-valuemin={this.min}
+                aria-valuemax={this.max}
+                aria-valuenow={this.value}
+                min={this.min}
+                max={this.max}
+                value={this.inputFieldValue}
+                onInput={(Event) => this.handleInputFieldChange(Event)}
+              />
+            </div>
+          </label>
+        </div>
+      </Host>
+    );
+  }
+
+  private setBackgroundProgressFill(): string {
+    const RELATIVE_POSITION =
+      ((this.value - this.min) / (this.max - this.min)) * 100;
+    return `linear-gradient(to right,
+      ${plmgColorBackgroundPrimaryStrong} 0%,
+      ${plmgColorBackgroundPrimaryStrong} ${RELATIVE_POSITION}%,
+      ${plmgColorBorderNeutralWeak} ${RELATIVE_POSITION}%,
+      ${plmgColorBorderNeutralWeak} 100%)`;
+  }
+
+  private setThumbPosition() {
+    const RELATIVE_POSITION = (this.value - this.min) / (this.max - this.min);
+
+    // Use the longest character in the range to set a min width. Prevents rapid shrink / expand as non-monospaced font varies width of the thumb label.
+    return {
+      minWidth: `calc(.5em * ${this.getLongestCharacterLength()}`,
+      transform: `translate(calc(${RELATIVE_POSITION}em * (${this.trackWidth} / 12 - 1.5) - 50%)`,
```

Comment:
This formula is exactly what the function `setLabelPosition()` returns. Give that function a better name for its two use cases and call it here.

Reply:
refactored into single function